src/PlaywrightSharp.Abstractions/WaitForNavigationOptions.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace PlaywrightSharp
+{
+    /// <summary>
+    /// Options for <see cref="IPage.WaitForNavigationAsync(WaitForNavigationOptions)"/> and <see cref="IFrame.WaitForNavigationAsync(WaitForNavigationOptions)"/>.
+    /// </summary>
+    public class WaitForNavigationOptions : NavigationOptions
+    {
+        /// <summary>
+        /// Wait for this specific URL.
+        /// </summary>
+        public string Url { get; set; }
+
+        /// <summary>
+        /// Wait for an URL matching this expression.
+        /// </summary>
+        public string UrlRegEx { get; set; }

```

Comment:
shouldn't this be included in `UrlPredicate`?

Reply:
Both `Url` and `UrlRegEx` will end up being a predicate, internally.